javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -172,7 +175,18 @@ final class Matcher private[regex] (
     lastMatch
   }
 
-  def groupCount(): Int = ensureLastMatch.length-1
+  def groupCount(): Int = {
+    lastGroupCount match {
+      case Some(n) => n
+
+      case None =>
+        val groupCountPattern = Pattern.compile("|" + pattern0.jsPattern)
+        val groupCountRegex = groupCountPattern.newJSRegExp()
```

Comment:
You can avoid the `Pattern.compile` step in this case. Just do
```scala
val groupCountRegex = new js.RegExp("|" + pattern0.jsPattern)
```
this will be faster. Flags are irrelevant, and `pattern0.jsPattern` has already been processed for in-pattern flags, so `Pattern.compile` is going to do work for nothing.